apps/emqx_management/src/emqx_mgmt_api_certs.erl

```diff
@@ -0,0 +1,343 @@
+            tags => ?TAGS,
+            description => ?DESC("global_file_upload"),
+            parameters => [param_path_bundle_name()],
+            'requestBody' => ref(file_in),
+            responses =>
+                #{
+                    204 => <<"">>,
+                    400 => bad_request(?DESC("bad_request")),
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        },
+        delete => #{
+            tags => ?TAGS,
+            description => ?DESC("global_bundle_delete"),
+            parameters => [param_path_bundle_name()],
+            responses =>
+                #{
+                    204 => <<"">>,
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        }
+    };
+schema("/certs/ns/:namespace/list") ->
+    #{
+        'operationId' => '/certs/ns/:namespace/list',
+        get => #{
+            tags => ?TAGS,
+            description => ?DESC("ns_bundle_list"),
+            parameters => [param_path_ns()],
+            responses =>
+                #{
+                    200 =>
+                        emqx_dashboard_swagger:schema_with_examples(
+                            array(ref(bundle_out)),
+                            example_bundle_list()
+                        ),
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        }
+    };
+schema("/certs/ns/:namespace/name/:name") ->
+    #{
+        'operationId' => '/certs/ns/:namespace/name/:name',
+        get => #{
+            tags => ?TAGS,
+            description => ?DESC("ns_file_list"),
+            parameters => [param_path_ns(), param_path_bundle_name()],
+            responses =>
+                #{
+                    200 =>
+                        emqx_dashboard_swagger:schema_with_examples(
+                            hoconsc:map(file_kind, ref(file_out)),
+                            example_file_list()
+                        ),
+                    404 => not_found(?DESC("bundle_not_found")),
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        },
+        post => #{
+            tags => ?TAGS,
+            description => ?DESC("ns_file_upload"),
+            parameters => [param_path_ns(), param_path_bundle_name()],
+            'requestBody' => ref(file_in),
+            responses =>
+                #{
+                    204 => <<"">>,
+                    400 => bad_request(?DESC("bad_request")),
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        },
+        delete => #{
+            tags => ?TAGS,
+            description => ?DESC("ns_bundle_delete"),
+            parameters => [param_path_ns(), param_path_bundle_name()],
+            responses =>
+                #{
+                    204 => <<"">>,
+                    500 => internal_error(?DESC("internal_error"))
+                }
+        }
+    }.
+
+fields(file_in) ->
```

Comment:
Maybe provide multipart/form-data endpoint?

Reply:
I followed the original proposal which is designed as a one-by-one upload.  multipart/form-data would be useful if we wanted to upload multiple files at once, right?

@zmstone WDYT?

Reply:
it's good to have multipart upload so one curl command or one request from dashboard can be used to upload multiple files.

I mean supporting multi-part does not mean one cannot upload single file anymore.